### PR TITLE
[SPARK-14844] [ML] Add setFeaturesCol and setPredictionCol to KMeansM…

### PR DESCRIPTION
**Pull request by dominik-jastrzebski**
## What changes were proposed in this pull request?

Introduction of setFeaturesCol and setPredictionCol methods to KMeansModel in ML library.
## How was this patch tested?

By running KMeansSuite.

### CODE REVIEW

**Comment by dominik-jastrzebski** [comment]
ok to test

**Comment by MLnick** [comment]
ok to test

File: mllib/src/test/scala/org/apache/spark/ml/clustering/KMeansSuite.scala [hunk 1]
```diff
@@ -117,6 +126,24 @@ class KMeansSuite extends SparkFunSuite with MLlibTestSparkContext with DefaultR
     assert(clusterSizes.forall(_ >= 0))
   }
 
+  test("transform with modified model's parameters") {
+    val newFeaturesColumnName = "kmeans_model_features"
```
Comment (MLnick) [inline mllib/src/test/scala/org/apache/spark/ml/clustering/KMeansSuite.scala]: I think it's enough to consolidate your two tests into one:

```scala
val featuresColName = "new_features"
val predictionColName = "new_prediction"
val model = new KMeans().setK(k).setSeed(1).fit(dataset)
model.setFeaturesCol(featuresColName).setPredictionCol(predictionColName)
val transformed = model.transform(dataset.withColumnRenamed("features", featuresColName))
Seq(featuresColName, predictionColName).foreach { col =>
  assert(transformed.columns.contains(column))
}
assert(model.getFeaturesCol === featuresColName)
assert(model.getPredictionCol === predictionColName)
```

**Comment by dominik-jastrzebski** [comment]
I made corrections suggested in code review.

File: mllib/src/test/scala/org/apache/spark/ml/clustering/KMeansSuite.scala [hunk 2]
```diff
@@ -117,6 +117,21 @@ class KMeansSuite extends SparkFunSuite with MLlibTestSparkContext with DefaultR
     assert(clusterSizes.forall(_ >= 0))
   }
 
+  test("transform with modified model's parameters") {
```
Comment (MLnick) [inline mllib/src/test/scala/org/apache/spark/ml/clustering/KMeansSuite.scala]: can we call this test "KMeansModel transform with non-default feature and prediction cols"

Comment (dominik-jastrzebski) [inline mllib/src/test/scala/org/apache/spark/ml/clustering/KMeansSuite.scala]: Done.

**Comment by MLnick** [comment]
LGTM, merged to master and branch-2.0. Thanks!

**Pull request closed without merging**